tests/test_rdump.py

```diff
@@ -461,6 +461,46 @@ def test_rdump_headerless_csv(tmp_path, capsysbinary):
     ]
 
 
+def test_rdump_stdin_peek(tmp_path):
+    if platform.system() == "Windows":
+        pytest.skip("No Gzip on Windows")
+
+    TestRecord = RecordDescriptor(
+        "test/record",
+        [
+            ("varint", "count"),
+            ("string", "foo"),
+        ],
+    )
+
+    path = tmp_path / "test.records"
+    writer = RecordWriter(path)
+    # generate some test records
+    for i in range(10):
+        writer.write(TestRecord(count=i, foo="bar"))
+    writer.close()
+
+    # Compress
+    compress_cmd = ["gzip", "/k", str(path)]
+
+    res = subprocess.Popen(compress_cmd, stdout=subprocess.PIPE)
+    res.communicate()
+
+    compressed_path = str(path) + ".gz"
```

Comment:
`gzip /k` should be `gzip -k/--keep`:

```suggestion
    # Gzip compress records file
    compress_cmd = ["gzip", "--keep", str(path)]
    subprocess.check_output(compress_cmd)
    compressed_path = str(path) + ".gz"
```